src/main/java/dev/openfga/sdk/api/configuration/ClientConfiguration.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * OpenFGA
+ * A high performance and flexible authorization/permission engine built for developers and inspired by Google Zanzibar.
+ *
+ * The version of the OpenAPI document: 0.1
+ * Contact: community@openfga.dev
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+package dev.openfga.sdk.api.configuration;
+
+import java.util.Optional;
+
+public class ClientConfiguration extends Configuration {
+    private String authorizationModelId;
+
+    public ClientConfiguration(String apiUrl) {
+        super(apiUrl);
+    }
+
+    public ClientConfiguration(String apiUrl, String authorizationModelId) {
```

Comment:
Can we not have all these iterations to start? (In both this and configuration) - this will get more confusing rather than less

Reply:
By iterations do you mean avoid constructor polymorphism?

I can update to a builder-only approach

Reply:
Yes, for the initial version and pending feedback let both configuration and clientconfiguration take the needed configs without the shortcuts.

Especially because the above can easily lead to confusing/breaking changes.

For example, if you were including store id, two of these cases would not work as expected:
ClientConfig("https://localhost:8080", "some-store-id")
ClientConfig("https://localhost:8080", "some-auth-model-id")
ClientConfig("https://localhost:8080", "some-store-id", "some-auth-model-id")
ClientConfig("some-store-id", "some-auth-model-id")